components/touchscreen/index.rst

```diff
@@ -19,28 +19,82 @@ Base Touchscreen Configuration
     # Example touchscreen
     touchscreen:
       - platform: ...
+        display: display1
         on_touch:
           then:
             ...
+        on_update:
+          then:
+            ...
+        on_release:
+          then:
+            ...
 
 Configuration variables:
 ************************
+- **display** (*Required*, :ref:`config-id`): The display to use this touchscreen with.
 
 - **on_touch** (*Optional*, :ref:`Automation <automation>`): An automation to perform
   when the touchscreen is touched. See :ref:`touchscreen-on_touch`.
-- **display** (**Required**, :ref:`config-id`): The display to use. If only one display is
-  available, this can be omitted.
+- **on_update** (*Optional*, :ref:`Automation <automation>`): An automation to perform
+  when the touchscreen is touched. See :ref:`touchscreen-on_update`.
+- **on_release** (*Optional*, :ref:`Automation <automation>`): An automation to perform
+  when the touchscreen is touched. See :ref:`touchscreen-on_release`.
+
+
+.. _touchscreen-touchpoint:
+
+``TouchPoint`` argument type.
+-----------------------------
+
+Both the :ref:`touchscreen-on_touch` and :ref:`touchscreen-on_update` have an argument of the type :apistruct:`touchscreen::TouchPoint` in a
+list or as variable (only ``on_touch``).
+
+The integer members for the touch positions below are in relation to the display width and height:
+
+- ``x`` and ``y`` this shows the current position.
+- ``x_last`` and ``y_last`` this shows the priveus position.
+- ``x_first`` and ``y_first`` This shows the position of when the touch was firstly detected.
+- ``x_raw`` and ``y_raw`` This if for calibrating the touchscreen in relation of the display. This replaces the properties with the same name in the touchscreen classes.
+
+- ``id`` is a number given by the touchscreen referencing the detected touch on a multi touch screen.
+- ``state`` shows the state of the touch. This can be **1** as just touched or **2** when the touched position has been moved around.
```

Comment:
```suggestion
- ``id`` is a number provided by the touchscreen to uniquely identify the touch on a multi-touch screen.
- ``state`` indicates the state of the touch. This can be **1**, indicating it is an initial touch, or **2** indicating the touch position has changed/moved.
```